src/lib/mina_lib/mina_lib.ml

```diff
@@ -1392,6 +1487,51 @@ let send_resource_pool_diff_or_wait ~rl ~diff_score ~max_per_15_seconds diff =
   able_to_send_or_wait ()
 
 let create ?wallets (config : Config.t) =
+  let _precomputed_block_writer_setup =
```

Comment:
If we're mutably doing the initialisation here, doesn't it make more sense to pass these values as an argument to the configuration?

Reply:
Yes. That sounds better.